app/models/user.rb

```diff
@@ -1,6 +1,34 @@
 class User < ActiveRecord::Base
+  mount_uploader :avatar, AvatarUploader
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+  # :authentication_keys => [:login]
+  attr_accessor :login
+
+  def self.find_for_database_authentication(warden_conditions)
+  # def self.find_first_by_auth_conditions(warden_conditions)
+    conditions = warden_conditions.dup
+    if login = conditions.delete(:login)
+      where(conditions).where(["lower(username) = :value OR lower(email) = :value", { :value => login.downcase }]).first
```

Comment:
this is also valid. you should have this line broken into two lines of syntax. 

Consider:

```ruby
where(conditions).where([
  "lower(username) = :value OR lower(email) = :value", 
  { :value => login.downcase }
]).first
```